ingestion/src/metadata/ingestion/source/snowflake.py

```diff
@@ -62,23 +65,31 @@ def get_connection_url(self):
 
 class SnowflakeSource(SQLSource):
     def __init__(self, config, metadata_config):
-        if config.connect_args.get("private_key"):
-            private_key = config.connect_args["private_key"]
-            p_key = serialization.load_pem_private_key(
-                bytes(private_key, "utf-8"),
-                password=os.environ["SNOWFLAKE_PRIVATE_KEY_PASSPHRASE"].encode(),
-                backend=default_backend(),
-            )
-            pkb = p_key.private_bytes(
-                encoding=serialization.Encoding.DER,
-                format=serialization.PrivateFormat.PKCS8,
-                encryption_algorithm=serialization.NoEncryption(),
+        config_dict = config.serviceConnection.__root__.config.dict()
```

Comment:
let's try to not convert to `dict`if possible. `connectionArguments` is an attribute of `SnowflakeConnection` which has `None` as default, so we should be good to go with:

```
connection_arguments = config.serviceConnection.__root__.config.connectionArguments
if connection_arguments:
    ...
```